src/logger/Logger.cpp

```diff
@@ -0,0 +1,194 @@
+#include "kompute/logger/Logger.hpp"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <mutex>
+#include <spdlog/async.h>
+#include <spdlog/common.h>
+#include <spdlog/logger.h>
+#include <spdlog/sinks/rotating_file_sink.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
```

Comment:
Why are these dependencies required?

Reply:
They are used for creating the logging directory and checking if it already exists.

Reply:
Ok I still thing there's a few things that are not clear on the reason for the helpers but would be good to get your thoughts on the comment above

Reply:
Removed this stuff like we have discussed.